Tests/UIKitTests/UINavigationControllerExtensionsTests.swift

```diff
@@ -53,5 +53,19 @@ final class UINavigationControllerExtensionsTests: XCTestCase {
         XCTAssertNotNil(color)
         XCTAssertEqual(color!, .red)
     }
+    
+    func testHideBottomBar() {
+        let rootVC = UIViewController()
+        let navigationController = UINavigationController(rootViewController: rootVC)
+        let vcToPush = UIViewController()
+        let tabVC = UITabBarController()
+        tabVC.viewControllers = [navigationController]
+        navigationController.pushViewController(vcToPush, hideBottomBar: true, animated: true)
+        XCTAssert(vcToPush.hidesBottomBarWhenPushed)
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1){
+            XCTAssert(tabVC.tabBar.isHidden)
+        }
+
```

Comment:
If you want to test something with a delay, you need to wait for an expectation.
Also, you probably don't even need the delay for the async call, if you set `animated: false`
```suggestion
        let expectation = XCTestExpectation(description: "Tab bar hidden")
        DispatchQueue.main.async {
            XCTAssert(tabVC.tabBar.isHidden)
            expectation.fulfill()
        }
        wait(for: [expectation], timeout: 0.1)
```

Reply:
Actually, if you don't animate then you may not need the `async` call altogether and can just run the test. Try that first.